plugins/custom_caret.lua

```diff
@@ -106,4 +119,28 @@ function DocView:draw_caret(x, y)
   end
 
   renderer.draw_rect(x, y, caret_width, caret_height, caret_color)
+  if current_caret_shape == "block" then
+    core.push_clip_rect(x, y, caret_width, caret_height)
+
+    local function draw_char(l, c)
+      l = common.clamp(l, 1, #self.doc.lines   )
+      c = common.clamp(c, 1, #self.doc.lines[l])
+      local cx,cy = self:get_line_screen_position(l, c)
+      renderer.draw_text(
+        font, self.doc:get_char(l, c),
+        cx, cy+self:get_line_text_y_offset(),
+        style.background
+      )
+    end
+
+    for yo=-1, 1 do
+      for xo=-1, 1 do
+        draw_char(line+xo, col+yo)
+      end
+    end
+
+    core.pop_clip_rect(x, y, caret_width, caret_height)
```

Comment:
`core.pop_clip_rect` doesn't take any parameter.

Reply:
I don't remember putting those there...

Reply:
Could you take'em out?